event-reader-sink/src/main/kotlin/gov/cdc/ocio/eventreadersink/cloud/AwsManager.kt

```diff
@@ -0,0 +1,82 @@
+package gov.cdc.ocio.eventreadersink.cloud
+
+import gov.cdc.ocio.eventreadersink.exceptions.ConfigurationException
+import mu.KotlinLogging
+import org.apache.camel.component.aws2.s3.AWS2S3Component
+import org.apache.camel.component.aws2.s3.AWS2S3Configuration
+import org.apache.camel.component.aws2.sqs.Sqs2Component
+import org.apache.camel.component.aws2.sqs.Sqs2Configuration
+
+private val logger = KotlinLogging.logger {}
+/**
+ * Class for creating Aws Components
+ */
+class AwsManager {
+
+    /**
+     * Configures and returns an instance of Sqs2Component for interacting with AWS SQS.
+     *
+     * This function initializes the SQS component with the provided AWS credentials
+     * and region, allowing for communication with the specified SQS service.
+     *
+     * @param awsAccessKeyId The AWS access key ID for authentication.
+     * @param awsSecretAccessKey The AWS secret access key for authentication.
+     * @param awsRegion The AWS region where the SQS service is located.
+     * @return An instance of Sqs2Component configured with the specified AWS credentials and region.
+     * @throws IllegalArgumentException
+     * @throws ConfigurationException
+     */
+    @Throws (IllegalArgumentException:: class, ConfigurationException:: class)
+    fun configureAwsSQSComponent(awsAccessKeyId: String, awsSecretAccessKey: String, awsRegion: String): Sqs2Component {
+
+        try {
+            val sqsComponent = Sqs2Component()
+            val sqsConfiguration = Sqs2Configuration()
+            sqsConfiguration.accessKey = awsAccessKeyId
+            sqsConfiguration.secretKey = awsSecretAccessKey
+            sqsConfiguration.region = awsRegion
+            sqsComponent.configuration = sqsConfiguration
+            return sqsComponent
+        } catch (e: IllegalArgumentException) {
+            logger.error("Invalid argument provided for SQS configuration: ${e.message}")
+            throw e
+        } catch (e: Exception) {
+            logger.error("Error configuring AWS SQS component: ${e.message}", e)
+            throw ConfigurationException("Failed to configure AWS SQS component: ${e.message}")
```

Comment:
If you are throwing ConfigurationException then use that to catch and then have another catch block for catch "all" exception